### PR TITLE
[two_dimensional_scrollables] Set same minimum version as rest of the repo

### PR DESCRIPTION
**Pull request by bparrishMines**
Changes the minimum Dart and Flutter versions as the rest of the repo. See https://github.com/flutter/packages/pull/4731 and  https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version

Our CI was unable to publish this plugin with a minimum Dart version of `3.1.0` since our repo runs the publish command with version `3.0.0`. See https://github.com/flutter/packages/actions/runs/5896017278/job/16010664124

```
Resolving dependencies...


The current Dart SDK version is 3.0.0.




Because two_dimensional_scrollables requires SDK version >=3.1.0 <4.0.0, version solving failed.


Publishing two_dimensional_scrollables failed.


The following packages had failures during publishing:
  packages/packages/two_dimensional_scrollables:
    publish failed
See above for full details.
Error: Process completed with exit code 1.
```

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by bparrishMines** [comment]
cc @stuartmorgan FYI

File: packages/two_dimensional_scrollables/example/pubspec.yaml [hunk 1]
```diff
@@ -6,8 +6,8 @@ publish_to: 'none'
 version: 1.0.0+1
 
 environment:
-  sdk: '>=3.1.0 <4.0.0'
-  flutter: ">=3.13.0"
+  sdk: '>=2.19.0 <4.0.0'
+  flutter: ">=3.7.0"
```
Comment (Piinks) [inline packages/two_dimensional_scrollables/example/pubspec.yaml:10]: This package won't work before Flutter 3.13
It uses new APIs that landed in 3.13

**Comment by bparrishMines** [comment]
Actually it looks like this won't work:

```
[0:00] Running for packages/two_dimensional_scrollables...
  The minimum Dart version is 2.19.0, but the minimum Flutter version of 3.13.0 shipped with Dart 3.1.0. Please use consistent lower SDK bounds
[0:00] Running for packages/two_dimensional_scrollables/example...
  The minimum Dart version is 2.19.0, but the minimum Flutter version of 3.13.0 shipped with Dart 3.1.0. Please use consistent lower SDK bounds
```

**Comment by bparrishMines** [comment]
This should be fixable by this instead: https://github.com/flutter/packages/pull/4745

**Pull request closed without merging**